stl/inc/ranges

```diff
@@ -7698,6 +7704,388 @@ namespace ranges {
         _EXPORT_STD inline constexpr _Zip_fn zip{};
     } // namespace views
 
+    _EXPORT_STD template <move_constructible _Func, input_range... _ViewTypes>
+        requires ((view<_ViewTypes> && ...) && (sizeof...(_ViewTypes) > 0)
+                  && is_object_v<_Func> && regular_invocable<_Func&, range_reference_t<_ViewTypes>...>
+                  && _Can_reference<invoke_result_t<_Func&, range_reference_t<_ViewTypes>...>>)
+    class zip_transform_view : public view_interface<zip_transform_view<_Func, _ViewTypes...>> {
+    private:
+        using _Inner_view = zip_view<_ViewTypes...>;
+
+        /* [[no_unique_address]] */ _Movable_box<_Func> _Function;
+        /* [[no_unique_address]] */ _Inner_view _Zip;
+
+        template <bool _IsConst>
+        using _Ziperator = iterator_t<_Maybe_const<_IsConst, _Inner_view>>;
+
+        template <bool _IsConst>
+        using _Zentinel = sentinel_t<_Maybe_const<_IsConst, _Inner_view>>;
+
+        template <bool _IsConst>
+        struct _Category_base {};
+
+        template <bool _IsConst>
+            requires forward_range<_Maybe_const<_IsConst, _Inner_view>>
+        struct _Category_base<_IsConst> {
+        private:
+            template <class _Tag_type>
+            static constexpr bool _Iterators_derive_from_tag =
+                (derived_from<
+                     typename iterator_traits<iterator_t<_Maybe_const<_IsConst, _ViewTypes>>>::iterator_category,
+                     _Tag_type>
+                    && ...);
+
+        public:
+            using iterator_category = decltype([]() noexcept {

```

Comment:
Marking decision needed for use of IILE.

Reply:
We talked about this at the weekly maintainer meeting, and decided that using immediately invoked lambda expressions for type computations (which is the case here) is reasonable, as it has no codegen impact, keeps the metaprogramming "local" to where it's needed (which is the usual reason to prefer lambdas for anything in the first place), and avoids the need to pass the template parameters (`_IsConst`, `_Func`, `_View_types`) to a helper struct. In this case, it's a relatively large lambda, but not so excessive as to impede clarity (and all of the lines are necessary). The alternative with a big stack of `conditional_t`s would be much worse to read.

We talked about a general guideline, and for runtime usage (i.e. outside of "manifestly constant-evaluated expressions"), we generally prefer to avoid immediately invoked lambda expressions, as there are potential throughput/codegen impacts, and we don't think that they usually improve clarity.